### PR TITLE
build: add clean script for all packages and workspace. Clean up dev …

### PR DESCRIPTION
**Pull request by cjpillsbury**
…dependencies. Don't build on install (prepare).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,12 +24,12 @@
     "shared/*"
   ],
   "scripts": {
+    "clean": "lerna run clean --parallel --scope @mux-elements/*",
     "format": "prettier --write .",
     "lint": "lerna run lint --scope @mux-elements/*",
     "test": "lerna run test --scope @mux-elements/*",
     "dev": "lerna run dev --parallel --scope @mux-elements/*",
     "build": "lerna run build",
-    "prepare": "husky install && yarn build --scope @mux-elements/*",
```
Comment (gkatsev) [inline package.json:32]: we probably want `yarn build --scope @mux-elements/*` on `prepublishOnly` or something

File: packages/mux-player/package.json [hunk 2]
```diff
@@ -43,6 +45,7 @@
     "eslint": "^7.11.0",
     "npm-run-all": "^4.1.5",
     "replace": "^1.2.1",
+    "shx": "^0.3.4",
```
Comment (gkatsev) [inline packages/mux-player/package.json:48]: Theoretically, we can have `shx` at the top level, and it'll be available for usage in all the workspace packages

File: package.json [hunk 3]
```diff
@@ -24,12 +24,12 @@
     "shared/*"
   ],
   "scripts": {
+    "clean": "lerna run clean --parallel --scope @mux-elements/*",
     "format": "prettier --write .",
     "lint": "lerna run lint --scope @mux-elements/*",
     "test": "lerna run test --scope @mux-elements/*",
     "dev": "lerna run dev --parallel --scope @mux-elements/*",
     "build": "lerna run build",
```
Comment (gkatsev) [inline package.json:32]: should we add `"prebuild": "clean"`

File: packages/playback-core/package.json [hunk 4]
```diff
@@ -14,6 +14,7 @@
   "author": "Mux, Inc.",
   "license": "MIT",
   "scripts": {
+    "clean": "shx rm -rf dist/",
     "dev:cjs": "open-process | yarn build:cjs --watch",
     "dev:types": "yarn build:types -w",
     "dev": "npm-run-all --parallel dev:types dev:cjs",
```
Comment (gkatsev) [inline packages/playback-core/package.json:20]: need to add `dev:esm` and add it in here

File: packages/mux-player-react/package.json [hunk 5]
```diff
@@ -34,6 +35,7 @@
     }
   },
   "dependencies": {
+    "@mux-elements/open-process": "0.1.0",
```
Comment (luwes) [inline packages/mux-player-react/package.json:38]: should be devDependencies

File: package.json (hunk 3 above)
Comment (cjpillsbury) [reply package.json:34]: Yeah it's out of scope but I am down. Gonna refactor my last version to just rely on your proposed prebuild as well.

File: packages/mux-player/package.json (hunk 2 above)
Comment (cjpillsbury) [reply packages/mux-player/package.json:48]: True. Philosophically, I prefer to add the dependencies where they are actually expected. It means version syncing/upgrades require some additional TLC but makes tracking and removing dependencies that are no longer used easier, and makes it clearer how dependencies are made available for where they're used.

**Review by gkatsev — APPROVED** [review]

File: packages/mux-player/package.json (hunk 2 above)
Comment (cjpillsbury) [reply packages/mux-player/package.json:49]: I'd lean towards using something like https://github.com/JamieMason/syncpack for those parts of housekeeping.

**Pull request merged**